File: GeneratedEntityReaderClasses/Components/LevitationComponent.cs
namespace GeneratedNoitaClasses;
public class LevitationComponent : NoitaComponentBase,  iNoitaType<LevitationComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaFloat radius { get; set; } = new();
	public NoitaFloat entity_force { get; set; } = new();
	public NoitaFloat box2d_force { get; set; } = new();
	public NoitaInt effect_lifetime_frames { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		radius.Read(s);
		entity_force.Read(s);
		box2d_force.Read(s);
		effect_lifetime_frames.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		radius.Write(s);
		entity_force.Write(s);
		box2d_force.Write(s);
		effect_lifetime_frames.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/LevitationComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaFloat
from BasicNoitaTypes import NoitaInt

from NoitaStream import NoitaStream
@dataclass
class LevitationComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    radius:NoitaFloat = field(default_factory=NoitaFloat)
    entity_force:NoitaFloat = field(default_factory=NoitaFloat)
    box2d_force:NoitaFloat = field(default_factory=NoitaFloat)
    effect_lifetime_frames:NoitaInt = field(default_factory=NoitaInt)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.radius.Read(s)
        self.entity_force.Read(s)
        self.box2d_force.Read(s)
        self.effect_lifetime_frames.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.radius.Write(s)
        self.entity_force.Write(s)
        self.box2d_force.Write(s)
        self.effect_lifetime_frames.Write(s)

